Fix #4410 - Unrendered font codes - plus add generic parser support for nestable callback MyCodes

lairdshaw opened the pull request:
Provide parser support for nestable font codes.

Resolves #4410

dvz commented:
Was the section moved to the other method because `postParser::$mycode_cache` doesn't support entries that are both `nestable` and `callback`? Would it be better to add this support?

lairdshaw commented:
> Was the section moved to the other method because `postParser::$mycode_cache` doesn't support entries that are both `nestable` and `callback`?

Yes, exactly.

> Would it be better to add this support?

Ideally, yes. I'll see what I can come up with.

lairdshaw committed:
Add generic parser support for nestable callback MyCodes

Convert special-case font MyCode handling into code that makes use of
the new generic support.

Sama34 commented:
I did test using the following post :
https://community.mybb.com/thread-226339-post-1342791.html#pid1342791

![image](https://user-images.githubusercontent.com/1786584/134248713-470bde8a-096f-41ae-b457-b11582a88e81.png)

There seem to be some issues, but these seem to be related to the `[url]` MyCode, so as far as I can assure it seems your code fixes the `[font]` issue.
